src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -184,22 +184,444 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        | BSE / Gaussian input (“coeffs x L²-norm prim”)  | 1.0              |
+        | Raw CRYSTAL deck (“coeffs for unnorm prim”)     | 1 / N_k          |
+        | Custom numeric STOs with internal scaling       | arbitrary        |
+
+        If omitted or a value of 1.0 is supplied, the primitives follow the
+        convention implied by `primitive_normalization`.  Only set a custom
+        factor when `primitive_normalization == 'custom'`.
+        """,
+    )
+
+    shell_normalization = Quantity(
+        type=np.float64,
+        description="""
+        **Global normalisation constant for the contracted shell** χ(r).
+
+        After primitives (and their `primitive_factor`s) are combined with the
+        contraction coefficients, one may optionally rescale the entire
+        contracted function so that
+
+        \\[
+            \\int_{\\mathbb{R}^3} |χ(r)|^2 \\,dr = 1.
+        \\]
+
+        - Many codes (Gaussian, ORCA, Q‑Chem) leave the contracted shell
+          *unnormalised* (value = 1.0).
+        - Other codes (GAMESS default) renormalise χ, in which case
+          `shell_normalization` equals the computed factor **1/√⟨χ|χ⟩**.
+        - Storing it makes the representation **loss‑free**: regardless of the
+          original convention, another program can reproduce exactly the same
+          contracted AOs by multiplying the linear combination *and then*
+          this scalar.
+
+        Leave at 1.0 (or None) if no extra shell‑level scaling is present.
+        """,
+    )
+
+    point_charge = Quantity(
+        type=np.float32,
+        description="""
+        If using a basis function that explicitly includes a point-charge or an 
+        ECP-like pseudo-component, this field can store that net charge. 
+        Typically 0 for standard GTO or STO expansions. 
+        Some extended basis concepts (or embedded charges) might set it differently.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        """
+        Canonicalise sizes and rebuild helper attributes.
+
+        * ``contraction_coefficients`` stays *flat* to satisfy the declared
+          1-D shape, but the property ``coeff_matrix`` (added below) provides
+          a reshaped **view** for convenience.
+        * ``coefficient_sets`` maps each angular‑momentum letter ('s','p',…)
+          to its slice in the flat list.
+        """
+        super().normalize(archive, logger)
+
+        # ---------------- infer component / primitive counts -------------
+        self.n_components = len(self.function_type) if self.function_type else 1
+        if self.n_primitive is None and self.exponents is not None:
+            self.n_primitive = len(self.exponents)
+
+        # ---------------- shape & size checks ----------------------------
+        if self.exponents is not None and len(self.exponents) != self.n_primitive:
+            raise ValueError('Length of exponents != n_primitive.')
```

Comment:
avoid using raise all throughout, simply log an error and return, we want to run the other normalizationsl